pybytes/pyscan/main.py

```diff
@@ -0,0 +1,86 @@
+# main.py -- put your code here!
+'''
+Simple Pyscan NFC / MiFare Classic Example
+Copyright (c) 2018, Pycom Limited.
+This example runs the NFC discovery loop in a thread.
+If a card is detected it will read the UID and compare it to VALID_CARDS
+RGB LED is BLUE while waiting for card,
+GREEN if card is valid, RED if card is invalid
+'''
+
+DEBUG = False  # change to True to see debug messages
+
+from pyscan import Pyscan
+from MFRC630 import MFRC630
+from LIS2HH12 import LIS2HH12
+from LTR329ALS01 import LTR329ALS01
+import binascii
+import time
+import pycom
+import _thread
+
+VALID_CARDS = [[0x43, 0x95, 0xDD, 0xF8],
+               [0x43, 0x95, 0xDD, 0xF9]]
+
+py = Pyscan()
+nfc = MFRC630(py)
+lt = LTR329ALS01(py)
+li = LIS2HH12(py)
+
+RGB_BRIGHTNESS = 0x8
+
+RGB_RED = (RGB_BRIGHTNESS << 16)
+RGB_GREEN = (RGB_BRIGHTNESS << 8)
+RGB_BLUE = (RGB_BRIGHTNESS)
+
+# Make sure heartbeat is disabled before setting RGB LED
+pycom.heartbeat(False)
+
+# Initialise the MFRC630 with some settings
+nfc.mfrc630_cmd_init()
+
+def check_uid(uid, len):
+    return VALID_CARDS.count(uid[:len])
+
+def print_debug(msg):
+    if DEBUG:
+        print(msg)
+
+def send_sensor_data(name, timeout):
+    while(pybytes):
+        pybytes.send_virtual_pin_value(True, 2, lt.light())
+        pybytes.send_virtual_pin_value(True, 3, li.acceleration())
+        time.sleep(timeout)
+
+def discovery_loop(arg1, arg2):
+    while True:
+        # Send REQA for ISO14443A card type
+        print_debug('Sending REQA for ISO14443A card type...')
+        atqa = nfc.mfrc630_iso14443a_WUPA_REQA(nfc.MFRC630_ISO14443_CMD_REQA)
+        print_debug('Response: {}'.format(atqa))
+        if (atqa != 0):
+            # A card has been detected, read UID
+            print_debug('A card has been detected, read UID...')
+            uid = bytearray(10)
+            uid_len = nfc.mfrc630_iso14443a_select(uid)
+            print_debug('UID has length: {}'.format(uid_len))
+            if (uid_len > 0):
+                print_debug('Checking if card with UID: [{:s}] is listed in VALID_CARDS...'.format(binascii.hexlify(uid[:uid_len],' ').upper()))
+                if (check_uid(list(uid), uid_len)) > 0:
+                    print_debug('Card is listed, turn LED green')
```

Comment:
You should send the card reads to Pybytes as well. (e.g. access denied for card {number}, or granted...)

Reply:
Currently added code to send card status.

Reply:
Also added to send uid to pybytes along with status, If no card attached added 0.